### PR TITLE
[Fix] Scale L2Wrap logits gradient by the upstream gradient

### PR DESCRIPTION
**Pull request by Sunt-ing**
## Summary

`L2Wrap.backward` returns the logits gradient without multiplying it by the upstream gradient, so any non-unit upstream gradient under-scales the L2 penalty gradient on the logits.

An `autograd.Function` must return input gradients that are linear in the incoming `grad_output`. `L2Wrap.backward` does this for the loss input (`return grad_output, ...`) but returns `glogits` directly for the logits input:

```python
glogits.scatter_(-1, ids, maxx)
return grad_output, glogits, None   # glogits is not scaled by grad_output
```

So when the wrapped loss is scaled through autograd, `dloss` scales but `dlogits` does not. The common `loss.backward()` path uses an implicit upstream gradient of 1 and is unaffected, which is why this was not caught. It surfaces for gradient accumulation that scales the loss, custom loss weighting, `torch.autograd.grad(..., grad_outputs=k)`, or any outer fused loss wrapper. `use_l2warp` is a shared training option across the FLA causal LMs (and defaults to `True` for RWKV7), so the affected surface is broad.

The fused path already does the right thing: `fused_linear_cross_entropy_backward` multiplies the whole gradient, including the L2Warp contribution, by the upstream `do`, and #874 aligned its normalization with standalone `l2_warp`. The two paths are meant to be equivalent but only agree at upstream gradient 1; standalone `l2_warp` is the one missing the scaling.

The fix multiplies the logits gradient by `grad_output`. It is a scalar for the wrapped loss, so it broadcasts:

```python
return grad_output, glogits * grad_output, None
```

Minimal repro (an autograd.Function's input gradients must scale linearly with the upstream gradient):

```python
import torch
from fla.modules.l2warp import l2_warp

logits = torch.randn(2, 64, 100, requires_grad=True)
loss = torch.randn(())
wrapped = l2_warp(loss.requires_grad_(), logits, 1.0)

g_loss_1, g_logits_1 = torch.autograd.grad(wrapped, (loss, logits), grad_outputs=torch.ones(()), retain_graph=True)
g_loss_3, g_logits_3 = torch.autograd.grad(wrapped, (loss, logits), grad_outputs=torch.full((), 3.0))

print("loss grad ratio  :", (g_loss_3 / g_loss_1).item())                          # 3.0
print("logits grad ratio:", (g_logits_3.abs().max() / g_logits_1.abs().max()).item())
# before: 1.0  (logits gradient ignores the upstream gradient)
# after : 3.0  (matches the loss gradient and the fused path)
```

## Test plan

A regression test was added to `tests/modules/test_l2warp.py` (`test_l2_warp_grad_output_scaling`): for a non-unit upstream gradient, both `dloss` and `dlogits` from `l2_warp` must scale linearly. It fails before the fix (logits ratio ~2x off) and passes after.

- `pytest tests/modules/test_l2warp.py::test_l2_warp_grad_output_scaling -q` passes with the fix; without it the `dlogits` assertion fails.
- `pytest tests/modules/test_l2warp.py -q` still passes, including the existing standalone-vs-fused equivalence test `test_fused_linear_cross_entropy_l2_warp`.
- `pre-commit run --files fla/modules/l2warp.py tests/modules/test_l2warp.py` is clean.

## Breaking changes

None.

### CODE REVIEW

File: fla/modules/l2warp.py [hunk 1]
```diff
@@ -37,7 +37,7 @@ def backward(ctx, grad_output):
         glogits = torch.zeros(ctx.logits_shape, device=grad_output.device,
                               dtype=grad_output.dtype)
         glogits.scatter_(-1, ids, maxx)
-        return grad_output, glogits, None
+        return grad_output, glogits * grad_output, None
```
[the remark this answers is not in the document]
Comment (Sunt-ing) [reply fla/modules/l2warp.py]: `l2_warp` is only ever called on the already-reduced scalar loss (the forward normalizes by `B*T`), so `grad_output` is a scalar and broadcasts fine. The previous code only accepted a
  non-scalar loss by ignoring `grad_output`, which is the bug this fixes.

**Review by yzhangcs — APPROVED** [review]
